docs/ownership_adv.rst

```diff
@@ -139,19 +139,107 @@ true global reference count.
+``shared_ptr``\s that share ownership with the existing one to be
+obtained for the same object using ``obj->shared_from_this()`` or
+``obj->weak_from_this()``.
+
+nanobind's support for ``std::enable_shared_from_this`` consists of three
+behaviors:
+
+* If a raw pointer ``ST *obj`` is returned from C++ to Python, and
+  there already exists an associated ``std::shared_ptr<ST>`` which
+  ``obj->shared_from_this()`` can locate, then nanobind will produce a
+  Python instance that shares ownership with it. The behavior is
+  identical to what would happen if the C++ code did ``return
+  obj->shared_from_this();`` (returning an explicit
+  ``std::shared_ptr<ST>`` to Python) rather than ``return obj;``.
+  The return value policy has no effect in this case; you will get
+  shared ownership on the Python side regardless of whether you used
+  ``rv_policy::take_ownership`` or ``rv_policy::reference``.
+
+  * Note that this behavior occurs only if such a ``std::shared_ptr<ST>``
+    already exists! If not, then nanobind behaves as it would without
+    ``enable_shared_from_this``: a raw pointer will transfer exclusive
+    ownership to Python by default, or will create a non-owning reference
+    if you use ``rv_policy::reference``.
+
+* If a Python object is passed to C++ as ``std::shared_ptr<ST> obj``,
+  and there already exists an associated ``std::shared_ptr<ST>`` which
+  ``obj->shared_from_this()`` can locate, then nanobind will produce a
+  ``std::shared_ptr<ST>`` that shares ownership with it: an additional
+  reference to the same control block, rather than a new control block
+  (as would occur without ``enable_shared_from_this``). This improves
+  performance and makes the result of ``shared_ptr::use_count()`` more
+  accurate.
+
+* If a Python object is passed to C++ as ``std::shared_ptr<ST> obj``, and
+  there is no associated ``std::shared_ptr<ST>`` that
+  ``obj->shared_from_this()`` can locate, then nanobind will produce
+  a ``std::shared_ptr<ST>`` as usual (with a new control block whose deleter
+  drops a Python object reference), *and* will do so in a way that enables
+  future calls to ``obj->shared_from_this()`` to find it as long
+  as any ``shared_ptr`` that shares this control block is still alive on
+  the C++ side.
+
+  (Once all of the ``std::shared_ptr<ST>``\s that share this control block
+  have been destroyed, the underlying PyObject reference being
+  managed by the ``shared_ptr`` deleter will be dropped,
+  and ``shared_from_this()`` will stop working. It can be reenabled by
+  passing the Python object back to C++ as ``std::shared_ptr<ST>`` once more,
+  which will create another control block.)
+
+Bindings for a class that supports ``enable_shared_from_this`` will be
+slightly larger than bindings for a class that doesn't, as nanobind
+must produce type-specific code to implement the above behaviors.
+
+.. warning:: The ``shared_from_this()`` method will only work when there
```

Comment:
Doesn't the body (`The [..]`) need to start on the line _following_ the `warning::` block? Maybe I just misremember.

Reply:
I've always written them this way, and the docs look correct when built. I think the only requirement is that continuation lines line up with the 'w' in `warning::`.